pkg/sriov/sriov.go

```diff
@@ -175,20 +175,13 @@ func (s *sriovManager) ApplyVFConfig(conf *sriovtypes.NetConf) error {
 		return fmt.Errorf("failed to lookup master %q: %v", conf.Master, err)
 	}
 	// 1. Set vlan
-	if conf.Vlan == nil {
-		vlan := new(int)
-		*vlan = 0
-		conf.Vlan = vlan
-	}
-	// set vlan qos if present in the config
-	if conf.VlanQoS != nil {
-		if err = s.nLink.LinkSetVfVlanQos(pfLink, conf.VFID, *conf.Vlan, *conf.VlanQoS); err != nil {
-			return fmt.Errorf("failed to set vf %d vlan configuration: %v", conf.VFID, err)
+	if conf.VlanProto != nil {
+		if err = s.nLink.LinkSetVfVlanQosProto(pfLink, conf.VFID, *conf.Vlan, *conf.VlanQoS, sriovtypes.VlanProtoInt[*conf.VlanProto]); err != nil {
+			return fmt.Errorf("failed to set vf %d vlan configuration - id %d, qos %d and proto %s: %v", conf.VFID, *conf.Vlan, *conf.VlanQoS, *conf.VlanProto, err)
 		}
 	} else {
-		// set vlan id field only
-		if err = s.nLink.LinkSetVfVlan(pfLink, conf.VFID, *conf.Vlan); err != nil {
-			return fmt.Errorf("failed to set vf %d vlan: %v", conf.VFID, err)
+		if err = s.nLink.LinkSetVfVlanQos(pfLink, conf.VFID, *conf.Vlan, *conf.VlanQoS); err != nil {
```

Comment:
this changes previous behaviour for vlan qos, where if not specified it would not set it.

we ok with this ?

Reply:
well, just to be on the safe side. If you compare these two functions in netlink (https://github.com/vishvananda/netlink/pull/888/files#diff-67266f0f3b975fc34cdf43a00db7317df2458a2b6ce3f028158652b9a45ff5ddR592 and https://github.com/vishvananda/netlink/pull/888/files#diff-67266f0f3b975fc34cdf43a00db7317df2458a2b6ce3f028158652b9a45ff5ddR623), they have different RT attributes (IFLA_VF_VLAN and IFLA_VF_VLAN_INFO) which have different checks. I just wanted to continue using what has been working for us when proto is not needed.

Having said that, it can be done of course. What do you think?

Reply:
lets simplify, they seem equivalent the latter adds in the vlan proto.

Reply:
done @adrianchiris. Please have a look again. Thanks.

Reply:
cool thx !